src/Analyzers/CSharp/Analyzers/RemoveUnusedParametersAndValues/CSharpRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs

```diff
@@ -93,6 +88,14 @@ protected override Location GetDefinitionLocationToFade(IOperation unusedDefinit
                 case DeclarationPatternSyntax declarationPattern:
                     return declarationPattern.Designation.GetLocation();
 
+                case RecursivePatternSyntax recursivePattern:
+                    Debug.Assert(recursivePattern.Designation is not null, "If we got to this point variable designation cannot be null");
+                    return recursivePattern.Designation!.GetLocation();
+
+                case ListPatternSyntax listPattern:
+                    Debug.Assert(listPattern.Designation is not null, "If we got to this point variable designation cannot be null");
```

Comment:
what ensured that?  can you call out where that check would have happened?

Reply:
I cannot point to a specific `if`, that checks for that. This is just die to the logic of an analyzer: it can report list and recursive pattern node only if it contains unused variable. Since variable in such cases is a `Designation` syntax, here this property must never be null. The IDE owns both the analyzer and codefix, so I guess, it is ok to rely on analyzer's behaviour in the codefix, right?